File: backend/src/main/java/com/tripmarket/domain/member/repository/MemberRepository.java
package com.tripmarket.domain.member.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.tripmarket.domain.member.entity.Member;
import com.tripmarket.domain.member.entity.Provider;

@Repository
public interface MemberRepository extends JpaRepository<Member, Long> {
	Optional<Member> findByProviderAndProviderId(Provider provider, String providerId);

	Optional<Member> findByEmail(String email);

	Optional<Member> findByEmailAndProvider(String email, Provider provider);
}

File: backend/src/main/java/com/tripmarket/domain/member/repository/MemberRepository.kt
package com.tripmarket.domain.member.repository

import com.tripmarket.domain.member.entity.Member
import com.tripmarket.domain.member.entity.Provider
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface MemberRepository : JpaRepository<Member, Long> {
    fun findByProviderAndProviderId(provider: Provider, providerId: String): Optional<Member>

    fun findByEmail(email: String): Optional<Member>
}
